cpp/velox/memory/VeloxMemoryPool.cc

```diff
@@ -143,11 +159,74 @@ class WrappedVeloxMemoryPool : public facebook::velox::memory::MemoryPool {
   }
 
   void free(void* FOLLY_NULLABLE p, int64_t size) {
-    auto alignedSize = sizeAlign<ALIGNMENT>(ALIGNER<ALIGNMENT>{}, size);
-    allocator_->free(p, alignedSize);
+    auto alignedSize = sizeAlign(size);
+    glutenAllocator_->free(p, alignedSize);
     release(alignedSize);
   }
 
+  bool allocateNonContiguous(
+      MachinePageCount numPages,
+      facebook::velox::memory::MemoryAllocator::Allocation& out,
+      MachinePageCount minSizeClass) {
+    if (!allocator_.allocateNonContiguous(
+            numPages,
+            out,
+            [this](int64_t allocBytes, bool preAllocate) {
+              if (memoryUsageTracker_ != nullptr) {
+                memoryUsageTracker_->update(preAllocate ? allocBytes : -allocBytes);
+              }
+            },
+            minSizeClass)) {
+      VELOX_CHECK(out.empty());
+      return false;
+    }
+    VELOX_CHECK(!out.empty());
+    VELOX_CHECK_NULL(out.pool());
+    out.setPool(this);
+    return true;
+  }
```

Comment:
If the methods don't actually call `glutenAllocator_`, can you leave some comments somewhere as FIXME? So we can continue the work once we have time.

Reply:
Sounds reasonable.